pkg/core/transaction/attribute.go

```diff
@@ -77,3 +79,34 @@ func (attr *Attribute) EncodeBinary(w io.Writer) error {
 	}
 	return fmt.Errorf("failed encoding TX attribute usage: 0x%2x", attr.Usage)
 }
+
+// Size returns the size in number bytes of the Attribute
+func (attr *Attribute) Size() int {
+	switch attr.Usage {
+	case ContractHash, ECDH02, ECDH03, Vote,
+		Hash1, Hash2, Hash3, Hash4, Hash5, Hash6, Hash7, Hash8, Hash9, Hash10, Hash11, Hash12, Hash13, Hash14, Hash15:
+		return 33 // uint8 + 32 = size(attrUsage) + 32
+	case Script:
+		return 21 // uint8 + 20 = size(attrUsage) + 20
+	case Description:
+		return 2 + len(attr.Data) // uint8 + uint8+ len of data = size(attrUsage) + size(byte) + len of data
+	default:
+		return 1 + len(attr.Data) // uint8 + len of data = size(attrUsage) + len of data
+	}
+}
+
+// MarshalJSON implements the json Marschaller interface
+func (attr *Attribute) MarshalJSON() ([]byte, error) {
+	j, err := json.Marshal(
+		struct {
+			Usage string `json:"usage"`
+			Data  string `json:"data"`
+		}{
+			attr.Usage.String(),
+			hex.EncodeToString(attr.Data),
+		})
```

Comment:
mb:

```go
	j, err := json.Marshal(map[string]string{
		"usage": attr.Usage.String(),
		"data": hex.EncodeToString(attr.Data),
	})
```

Reply:
I don’t think that we really need a structure for that

Reply:
:+1: